### DISCUSSION TITLE
[4.x]: New naming convention for user permissions is very unclear

### DISCUSSION THREAD

**Discussion by darylknight** [craftcms/cms#13375] [Ideas] [answered]
I'm setting up more user groups for a Craft 4 site, and I've just seen that the labels for permissions have been changed. In Craft 3, you could set up a user group which allowed a user to create entries and save changes, but they couldn't make those changes live and required a user with the "Publish live changes" permission to actually save the entry.

I find the new naming convention for the ability to "Save entries" very unclear. Please consider changing it back to how it used to be, "Publish live changes".

- "Save Entries" is something that everyone is always able to do, because if they can Edit an entry, then they save their changes, even if the thing they save is turned into a Draft. It's not clear from "Save Entries" that they unable to "save them and make their saved changes live"
- "Publish live changes" is clearer that you can edit an entry and save your changes, but you can't make them live.

Old: 
<img width="372" alt="Old entry permissions" src="https://github.com/craftcms/cms/assets/1250017/424e76bf-3c33-4299-871f-7a8804524e16">

New:
<img width="260" alt="New entry permissions" src="https://github.com/craftcms/cms/assets/47792/5a389a74-ece6-4271-b051-e92db2baf46d">

### Craft CMS version

4.4.14

**Comment by brandonkelly** [comment]
The permission changes were made via #10467. Maybe this helps clear it up?

> (There’s no longer a distinction between saving an enabled or disabled entry, which is what “publish” used to refer to here. Nowadays if a user wants to make a non-live change to an element, they can create a draft.)

**Comment by darylknight** [comment]
I guess closing it means it won't be changed, but no - I still think the wording is confusing. When you're creating new user groups, I think it's now very unclear how you can set up a group that is able to edit, but not publish their changes.

**Comment by MoritzLost** [comment]
@brandonkelly For what it's worth, I agree with @darylknight – the new terminology is really confusing. Whenever I work on a site with granular permissions, I have to sift through the documentation and old issues, as well as do some trial and error.

Most confusing to me is the following:

1. _Save entries_ really means _Publish a draft_, since users can always create drafts as long as they have _View_ permission.
2. _View entries_ allows you to view entries but **also** to create drafts for existing entries. Creating drafts could be a separate permission – though I guess the use-case for being able to view entries in the backend without being able to do anything to them and is pretty rare.
3. _Create entries_ only allows users to create drafts, **unless** they also have _Save_ permission. A bit strange that creating an unpublished draft for a new entry is a separate permission, while creating a draft for an existing entry is not. But it doesn't hurt either way.

I think 2 & 3 are fine, only the fact that drafts for existing and new entries are treated differently makes it a bit confusing what the _Create entries_ permission does. But the real issue is that _Save_ really means _Publish_ – the previous wording in 3.x was much clearer in that sense.

I would suggest the following naming changes (no BC-break, the underlying permissions can stay the same, only the wording would be improved):

- _Save entries_ -> _Publish drafts_
- _Create entries_ -> _Create drafts_

And the _View_ permission could be made more explicit as well (a bit verbose, maybe that's a case for making this into two separate permissions):

- _View entries_ -> _View entries and create drafts for existing entries_

**Comment by brandonkelly** [comment]
> Save entries really means Publish a draft

Sortof. “Publish” implies publishing to the front-end, which only happens if the entry is enabled. Which is why we stopped using that word in Craft 4 – since there’s no longer a differentiation between saving disabled entries and enabled entries.

We’re discussing this internally and considering some ways to improve. Perhaps:

- View entries → View existing entries and create drafts
- Create entries → Create drafts of new entries

(and leave “Save entries” alone)

<img width="347" alt="User permissions with the above label changes" src="https://github.com/craftcms/cms/assets/47792/441c842f-6848-432c-ba7a-37d7d224c307"><br>


But now that I think about it, it’s a little awkward that we are differentiating between creating drafts of existing entries and creating drafts of new entries. So maybe it makes more sense for us to just combine those into a single “Create drafts” permission:

<img width="311" alt="User permissions with “Create entries” replaced by “Create drafts”" src="https://github.com/craftcms/cms/assets/47792/d4b0f8c4-c878-488e-ba66-e20c7e9d735e">

**Comment by MoritzLost** [comment]
@brandonkelly That sounds great!

> Sortof. “Publish” implies publishing to the front-end, which only happens if the entry is enabled. Which is why we stopped using that word in Craft 4 – since there’s no longer a differentiation between saving disabled entries and enabled entries.

Good to know the reasoning! But is it important to have that distinction in the wording of the permission? After all, users with that permission can always enable a disabled entry. And in some cases, disabled entries could still be visible in the frontend. What really matters here, I think, is that users can update/create the live/published/canonical version of a published entry (technically not _canonical_ since unpublished drafts are also canonical). In the end, the permission allows users to update/create a version of an entry that _may_ be visible in the frontend – in that sense, I still think _publish_ is a better wording than _save_.

> But now that I think about it, it’s a little awkward that we are differentiating between creating drafts of existing entries and creating drafts of new entries. So maybe it makes more sense for us to just combine those into a single “Create drafts” permission:

I think that makes perfect sense! That stood out to me as well when I wrote my comment above, though I didn't want to suggest this change since I wasn't sure if it would be a BC break 😅 It might be, in case some frontend code relies on users with the _View_ permission to be able to create drafts for existing entries.

In the meantime, I think the wording changes you mentioned are a solid improvement 👍 Though I would still suggest a change to the _Save_ permission, that one is the least clear in my opinion.

**Comment by brandonkelly** [comment]
> But is it important to have that distinction in the wording of the permission?

This whole conversation is about confusing permission labels, so yes, I would say avoiding a label that implies something more specific than it’s actually for is important :)

**Comment by MoritzLost** [comment]
@brandonkelly Absolutely! Though I would still argue that _Save_ is more confusing than _Publish_, even given the nuances of what _Publish_ might mean, since _Save_ can also mean a bunch of things (like saving drafts).

Maybe there's another wording that better captures the difference between creating drafts and creating something that the public might see?

**Comment by mmikkel** [comment]
> Create entries → Create drafts

This would be a very good call IMO, especially considering the editing UI uses "Create entry" for when a draft is first made into a real entry (an action that actually requires the "Save entries" permission).

> “Publish” implies publishing to the front-end, which only happens if the entry is enabled.

> I would still argue that Save is more confusing than Publish

"Save entries" is a bit fuzzy because it's technically required for (what is often perceived to be) a few different actions, like applying drafts, saving existing entries, and creating new entries from drafts. It's probably going to be difficult to find a name that accurately describes everything it does, though (at least without getting a bit verbose).

Maybe instead of renaming the permission, it could be given an info toggle (similar to the one a few of the user permissions have), explaining in a bit more detail what it does?

![CleanShot 2023-06-30 at 18 00 23](https://github.com/craftcms/cms/assets/298510/b52bfe0a-c045-40e7-ba64-fa8022b55d50)

**Comment by brandonkelly** [comment, marked as answer]
> Maybe instead of renaming the permission, it could be given an info toggle (similar to the one a few of the user permissions have), explaining in a bit more detail what it does?

This is the direction we decided to take for Craft 4, and will consider adding the explicit “Create drafts” permission in Craft 5.

Craft 4.4.16 was just released, which adds the following info messages:

- **View entries**: Allows viewing existing entries and creating drafts for them.
- **Create entries**: Allows creating drafts of new entries.
- **Save entries**: Allows fully saving canonical entries (directly or by applying drafts).
